specification/operationalinsights/data-plane/Microsoft.OperationalInsights/preview/2021-05-19_Preview/OperationalInsights.json

```diff
@@ -0,0 +1,1399 @@
+      "type": "string"
+    },
+    "workspacesParam": {
+      "description": "Workspace names to include in cross-workspace queries.",
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    },
+    "queryBody": {
+      "description": "The Analytics query. Learn more about the [Analytics query syntax](https://azure.microsoft.com/documentation/articles/app-insights-analytics-reference/)",
+      "type": "object",
+      "properties": {
+        "query": {
+          "description": "The query to execute.",
+          "$ref": "#/definitions/queryParam"
+        },
+        "timespan": {
+          "description": "Optional. The timespan over which to query data. This is an ISO8601 time period value.  This timespan is applied in addition to any that are specified in the query expression.",
+          "$ref": "#/definitions/timespanParam"
+        },
+        "workspaces": {
+          "description": "A list of workspaces that are included in the query.",
+          "$ref": "#/definitions/workspacesParam"
+        }
+      },
+      "required": [
+        "query"
+      ]
+    },
+    "queryResults": {
+      "title": "A query response.",
+      "description": "Contains the tables, columns & rows resulting from a query.",
+      "type": "object",
+      "properties": {
+        "tables": {
+          "description": "The list of tables, columns and rows.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/table"
+          }
+        },
+        "statistics": {
+          "type": "object",
+          "description": "Statistics represented in JSON format.",
+          "properties": {}
+        }
+      },
+      "required": [
+        "tables"
+      ]
+    },
+    "table": {
+      "title": "A query response table.",
+      "description": "Contains the columns and rows for one table in a query response.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the table.",
+          "type": "string"
+        },
+        "columns": {
+          "description": "The list of columns in this table.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/column"
+          }
+        },
+        "rows": {
+          "description": "The resulting rows from this query.",
+          "type": "array",
+          "items": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            }
+          }
+        }
+      },
+      "required": [
+        "name",
+        "columns",
+        "rows"
+      ]
+    },
+    "column": {
+      "title": "A table column.",
+      "description": "A column in a table.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of this column.",
+          "type": "string"
```

Comment:
Is the name optional?

Reply:
It shouldn't be but all the previous versions of the swagger didn't declared this field as `required`, so, didn't make it more restrictive than the previous versions.

Reply:
Can we clarify this? It would have API/functional consequences for libraries.